File: app/src/main/java/vk/kirisaki/service/RentService.java
package vk.kirisaki.service;

import java.time.LocalDateTime;
import java.util.List;

import vk.kirisaki.models.RentableObject;

public class RentService {
    private List<RentableObject> rentableObjects = List.of(
        new RentableObject("Car"),
        new RentableObject("Plates")
    );

    public List<RentableObject> getRentableObjects() {
        return rentableObjects;
    }

    public List<RentableObject> getAvailableRentableObjects(LocalDateTime dateTime) {
        return rentableObjects.stream().filter(o -> o.isAvailable(dateTime)).toList();
    }
}

File: src/rent_service.go
package main

import (
	"time"
)

type RentService struct {
	RentableObjects []RentableObject
}

func NewRentService() *RentService {
	return &RentService{
		RentableObjects: []RentableObject{
			{Name: "Car"},
			{Name: "Plates"},
		},
	}
}

func (rs *RentService) GetAvailableObjects(atTime time.Time) []RentableObject {
	var available []RentableObject
	for _, obj := range rs.RentableObjects {
		if obj.IsAvailable(atTime) {
			available = append(available, obj)
		}
	}
	return available
}
